src/elements/content-sidebar/versions/VersionsItem.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @flow
+ * @file Versions Item component
+ * @author Box
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+import { FormattedMessage } from 'react-intl';
+import { generatePath, withRouter } from 'react-router-dom';
+import type { Match } from 'react-router-dom';
+import DateField from '../../common/date';
+import messages from './messages';
+import NavButton from '../../common/nav-button';
+import sizeUtil from '../../../utils/size';
+import {
+    PLACEHOLDER_USER,
+    VERSION_DELETE_ACTION,
+    VERSION_RESTORE_ACTION,
+    VERSION_UPLOAD_ACTION,
+} from '../../../constants';
+import './VersionsItem.scss';
+
+type Props = {
+    match: Match,
+} & BoxItemVersion;
+
+const ACTION_MAP = {
+    [VERSION_DELETE_ACTION]: messages.versionRemovedBy,
+    [VERSION_RESTORE_ACTION]: messages.versionRestoredBy,
+    [VERSION_UPLOAD_ACTION]: messages.versionUploadedBy,
+};
+
+const DATE_FORMAT = {
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+    month: 'short',
+};
+
+const getActionMessage = action => (ACTION_MAP[action] ? ACTION_MAP[action] : ACTION_MAP[VERSION_UPLOAD_ACTION]);
+
+const VersionsItem = ({
+    action = VERSION_UPLOAD_ACTION,
+    id: versionId,
+    match,
+    modified_at: modifiedAt,
+    modified_by: modifiedBy = PLACEHOLDER_USER,
+    size,
+    version_number: versionNumber,
+}: Props) => {
+    const isDeleted = action === VERSION_DELETE_ACTION;
+    const className = classNames('bcs-VersionsItem', {
+        'bcs-is-disabled': isDeleted,
+    });
+    const versionPath = generatePath(match.path, { ...match.params, versionId });
+    const versionUser = modifiedBy.name || <FormattedMessage {...messages.versionUserUnknown} />;
+
+    return (
+        <NavButton
```

Comment:
If i'm thinking of the right component, clicking on this adds the selected style but doesn't actually navigate you anywhere? Do we need to update our route if we haven't previewed it yet?

Reply:
The NavButton is very similar to react-router's NavLink, just using a button instead of an anchor. Clicking on one of these does update the history route in memory and will trigger Preview to load the selected version. The NavButton will then see its `to` prop matches the current route and automatically apply its `activeClassName` to itself.